### ISSUE TITLE
Test step sub-field ? 

### ISSUE THREAD

**Issue by rafael5g** [jonasclaes/playwright-testmo-reporter#8]
Hello, loving using the Testmo Reporter for Playwright, just wanted to know how it's possible to add to the report test subfields ?

<test-steps>    
    <test-step status="passed">
      <test-step-subfield name="Step">
        This is the <em>description</em>.
      </test-step-subfield>
      <test-step-subfield name="Expected">
        And the <em>expected results</em>.
      </test-step-subfield>
    </test-step>

    <test-step status="failed">
      <test-step-subfield name="Step">
        This is the <em>description</em>.
      </test-step-subfield>
      <test-step-subfield name="Expected">
        And the <em>expected results</em>.
      </test-step-subfield>
    </test-step>
</test-steps>

If not possible, can this be a new feature request please ?

**Comment by jonasclaes** [comment]
Hello @rafael5g!

Thanks for logging this issue.

I assume you're referring to the following? https://docs.testmo.com/docs/automation/advanced/html-reference#example-test-steps

![image](https://github.com/jonasclaes/playwright-testmo-reporter/assets/25551249/f987347d-b2d7-468c-8304-568e1407d476)

This is implemented differently in this package.  
To enable this functionality, change the config of the reporter to include test steps like so:
```typescript:
...
reporter: [
    [
      'playwright-testmo-reporter',
      {
        embedTestSteps: true, // Optional: Embed test steps in the XML file. Defaults to true.
      }
    ]
]
...
```

Does this answer your question?

Kind regards,

**Comment by rafael5g** [comment]
Hello @jonasclaes , thanks for the quick reply.
Sorry if I wasn't clear, I have this option already on my playwright_config file : 

`reporter: [   
    ["line"],
    [
      "playwright-testmo-reporter",
      {
        outputFile: "testmo/chromium/testmo.xml", // Optional: Output file path. Defaults to 'testmo.xml'.
        embedTestSteps: true, // Optional: Embed test steps in the XML file. Defaults to true.
        //testStepCategories: ["hook","expect","pw:api","test.step"], // Optional: Test step categories to include in the XML file. Defaults to ["hook","expect","pw:api","test.step"]. Possible options are "hook", "expect", "pw:api", "test.step".
        testStepCategories: ["test.step"], // Optional: Test step categories to include in the XML file. Defaults to ["hook","expect","pw:api","test.step"]. Possible options are "hook", "expect", "pw:api", "test.step".
        testTitleDepth: 2, // Optional: Test case title depth to report in the XML file. Defaults to 1. Increase this to 2 include suite name. Increase this even further to include the path.
      },
    ],   
  ],`
  
  I wanted to add a subfield to my test step just like this example : 
  
![TestmoSubField](https://github.com/jonasclaes/playwright-testmo-reporter/assets/151641331/7f7617b2-4090-43cf-b4a4-942f6e2cec7f)

I tried to add a dubstep like this and didn't work:
`await test.step("First step", async () => {
await test.step("Sub step with Expected condition", async () => {
});
});`

Maybe I'm doing something wrong, but I couldn't create a substep to add an expected condition for example.

**Comment by jonasclaes** [comment]
Hello @rafael5g,

I see what you're trying to do. At this point in time, steps in steps are not supported by this reporter.
I'll turn this issue into a feature request.

Timing wise, is this a blocker for you at the moment, or rather a nice to have?

Kind regards,

**Comment by jonasclaes** [comment]
Hi @rafael5g,

Thanks for informing me! I'll get started on this.
Soon, I'll release a new beta version. Would you like to try this out then?

Kind regards

**Comment by rafael5g** [comment]
Yes sure, let me know please when I can try :)

Best regards.

**Comment by jonasclaes** [comment]
Hello @rafael5g,

I've added these changes. Can you try out this new build?

https://www.npmjs.com/package/playwright-testmo-reporter/v/1.8.5-add-substepping.0

You can install it by running:
`npm i playwright-testmo-reporter@1.8.5-add-substepping.0`
or
`npm i playwright-testmo-reporter@add-substepping`

Make sure to change the `includeTestSubFields` value to `true` in order to enable this new functionality.
![image](https://github.com/jonasclaes/playwright-testmo-reporter/assets/25551249/57487511-180d-49e0-89f5-b2a6cfaec964)

Steps are automatically correctly reported.
Expects are also supported, but if you want meaningful names in your reporting, you should add a custom text to each expect. For example:
```typescript
await expect(locator).toBeVisible() // This will report as "expect.toBeVisible"
await expect(locator, "Expect heading Projects to be visible").toBeVisible() // This will report as "Expect heading Projects to be visible"
```

Let me know what you think!

Kind regards

**Comment by rafael5g** [comment]
Hello @jonasclaes ,
Exactly what I was looking for, and the implementation on all the categories was an awesome idea.

Tested a bit and exported to Testmo, looks good.
Thanks once again for the help and support.

Best regards.

**Comment by jonasclaes** [comment]
Hey @rafael5g,

That's great to hear! 

It's now officially released as [v1.9.0](https://github.com/jonasclaes/playwright-testmo-reporter/releases/tag/v1.9.0).
Please update your `package.json` accordingly.

Kind regards
